service/history/timer_queue_active_task_executor.go

```diff
@@ -500,7 +505,8 @@ func (t *timerQueueActiveTaskExecutor) executeWorkflowTimeoutTask(
 		return err
 	}
 	if mutableState == nil || !mutableState.IsWorkflowExecutionRunning() {
-		return nil
+		release(nil) // release(nil) so mutable state doesn't get unloaded from cache
+		return consts.ErrWorkflowExecutionNotFound
```

Comment:
nit: just double checking that we don't have to touch this task type in this PR?

Reply:
Right. I had added it before I noticed we are not supporting ContinueAsNew via timeout action. I will remove it for now since it is not necessary.